include/AdePT/core/AdePTTransport.h

```diff
@@ -109,21 +86,23 @@ class AdePTTransport {
   GPUstate *fGPUstate{nullptr};                ///< CUDA state placeholder
   AdeptScoring *fScoring{nullptr};             ///< User scoring object
   AdeptScoring *fScoring_dev{nullptr};         ///< Device ptr for scoring data
-  static G4HepEmState *fg4hepem_state;         ///< The HepEm state singleton
+  G4HepEmState *fg4hepem_state{nullptr};       ///< The HepEm state singleton
   TrackBuffer fBuffer;                         ///< Vector of buffers of tracks to/from device (per thread)
-  bool fTrackInAllRegions;                     ///< Whether the whole geometry is a GPU region
   std::vector<std::string> *fGPURegionNames{}; ///< Region to which applies
   IntegrationLayer fIntegrationLayer; ///< Provides functionality needed for integration with the simulation toolkit
+  bool fInit{false};                  ///< Service initialized flag
+  bool fTrackInAllRegions;            ///< Whether the whole geometry is a GPU region
 
   /// @brief Used to map VecGeom to Geant4 volumes for scoring
   void InitializeSensitiveVolumeMapping(const G4VPhysicalVolume *g4world, const vecgeom::VPlacedVolume *world);
   void InitBVH();
   void InitializeUserData() { fScoring->InitializeOnGPU(); }
+  bool InitializeField(double bz);
   bool InitializeGeometry(const vecgeom::cxx::VPlacedVolume *world);
   bool InitializePhysics();
-  void InitializeGPU();
-  void ShowerGPU(int event, TrackBuffer &buffer); // const &buffer);
-  void FreeGPU();
+  void ProcessGPUHits();
```

Comment:
This method is declared in the integration layer

Reply:
Yes, this should go away as it is undefined, it doesn't cry just because it is not used in a templated class